### DISCUSSION TITLE
Using the "as" prop and typescript

### DISCUSSION THREAD

**Discussion by markwatabe** [chakra-ui/chakra-ui#4310] [Q&A]
I have implemented a render prop that expects an HTMLButtonElement. In theory`<HStack as="button" ...` should work, but ts doesnt pick up on the as prop meaning a button will be rendered. Does anyone have an idea how to resolve this?

**Comment by chasinhues** [comment]
Have you had a look at https://chakra-ui.com/guides/as-prop?
